src/gc-mmtk.c

```diff
@@ -0,0 +1,1198 @@
+#include "gc-common.h"
+#include "mmtkMutator.h"
+#include "threading.h"
+
+// File exists in the binding
+#include "mmtk.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// ========================================================================= //
+// Julia specific
+// ========================================================================= //
+
+extern jl_value_t *cmpswap_names JL_GLOBALLY_ROOTED;
+extern const unsigned pool_sizes[];
+extern void _jl_free_stack(jl_ptls_t ptls, void *stkbuf, size_t bufsz);
+extern jl_mutex_t finalizers_lock;
+
+// FIXME: Does it make sense for MMTk to implement something similar
```

Comment:
Something similar to what?

Reply:
Addressing that as part of the comment below in https://github.com/JuliaLang/julia/pull/56288#discussion_r1870414675 by changing `full_sweep_reasons` in `timing.jl`.